terraform/modules/environment/iam.tf

```diff
@@ -0,0 +1,93 @@
+
+# Data sources for EKS IAM
+data "aws_caller_identity" "current" {}
+
+# @TODO - sort out creating only a single user but multiple roles per env
+
+# Create KubernetesAdmin role for aws-iam-authenticator
+resource "aws_iam_role" "kubernetes_admin_role" {
+  name               = "${var.project}-kubernetes-admin-${var.environment}"
+  assume_role_policy = data.aws_iam_policy_document.assumerole_root_policy.json
+  description        = "Kubernetes administrator role (for AWS EKS auth)"
+}
+
+# Trust relationship to limit access to the k8s admin serviceaccount
+data "aws_iam_policy_document" "assumerole_root_policy" {
+  statement {
+    actions = ["sts:AssumeRole"]
+
+    principals {
+      type        = "AWS"
+      identifiers = ["arn:aws:iam::${data.aws_caller_identity.current.account_id}:root"]
+    }
+  }
+
+  # Allow the CI user to assume this role
+  statement {
+    actions = ["sts:AssumeRole"]
+
+    principals {
+      type        = "AWS"
+      identifiers = [data.aws_iam_user.ci_user.arn]
```

Comment:
it can access the `ci_user` from the main file right?

Reply:
Yes, since they are in the same directory. Though it looks like I forgot to remove something from main that I copied over. Thanks!